packages/contracts/test/chugsplash/hardhat-tools.spec.ts

```diff
@@ -0,0 +1,326 @@
+import { expect } from '../setup'
+
+/* Imports: External */
+import hre from 'hardhat'
+import { ethers } from 'ethers'
+import { remove0x } from '@eth-optimism/core-utils'
+
+/* Imports: Internal */
+import {
+  ChugSplashActionType,
+  getContractDefinition,
+  makeActionBundleFromConfig,
+} from '../../src'
+import { NON_NULL_BYTES32, NON_ZERO_ADDRESS } from '../helpers'
+
+describe('ChugSplash hardhat tooling', () => {
+  describe('makeActionBundleFromConfig', () => {
+    it('shoulld make a bundle from config with one contract and no variables', async () => {
+      const bundle = await makeActionBundleFromConfig(hre, {
+        contracts: {
+          MyContract: {
+            address: `0x${'11'.repeat(20)}`,
+            source: 'Helper_StorageHelper',
+            variables: {},
+          },
+        },
+      })
+
+      expect(bundle.actions.length).to.equal(1)
+      expect(bundle.actions[0].action).to.deep.equal({
+        actionType: ChugSplashActionType.SET_CODE,
+        target: `0x${'11'.repeat(20)}`,
+        data: getContractDefinition('Helper_StorageHelper').deployedBytecode,
+      })
+    })
+
+    it('should make a bundle from config with two contracts and no variables', async () => {
+      const bundle = await makeActionBundleFromConfig(hre, {
+        contracts: {
+          MyContract1: {
+            address: `0x${'11'.repeat(20)}`,
+            source: 'Helper_StorageHelper',
+            variables: {},
+          },
+          MyContract2: {
+            address: `0x${'22'.repeat(20)}`,
+            source: 'Helper_StorageHelper',
+            variables: {},
+          },
+        },
+      })
+
+      expect(bundle.actions.length).to.equal(2)
+      expect(bundle.actions[0].action).to.deep.equal({
+        actionType: ChugSplashActionType.SET_CODE,
+        target: `0x${'11'.repeat(20)}`,
+        data: getContractDefinition('Helper_StorageHelper').deployedBytecode,
+      })
+      expect(bundle.actions[1].action).to.deep.equal({
+        actionType: ChugSplashActionType.SET_CODE,
+        target: `0x${'22'.repeat(20)}`,
+        data: getContractDefinition('Helper_StorageHelper').deployedBytecode,
+      })
+    })
+
+    it('should make a bundle from config with one contract with variables', async () => {
+      const bundle = await makeActionBundleFromConfig(hre, {
+        contracts: {
+          MyContract1: {
+            address: `0x${'11'.repeat(20)}`,
+            source: 'Helper_StorageHelper',
+            variables: {
+              _uint8: 123,
```

Comment:
Could we consider `variables` being an array of `{ variableName: string, value: SolidityValue }` instead? Dynamic key names are always bug-prone.

Reply:
Building on this thread, it may also be that we want to define a specific order for some reason.  One example is if we are updating a code path in such a way that would change how transactions are ingested, and we want to apply all other update transactions with the old code path, and have the switch to the new one be the final step of the upgrade.